lib/dangers_detector.ex

```diff
@@ -61,7 +64,14 @@ defmodule ExcellentMigrations.DangersDetector do
   """
   @spec detect_dangers(ast, String.t()) :: [{danger_type, line}]
   def detect_dangers(ast, source_code) do
-    parsed_dangers = AstParser.parse(ast)
+    parsed_dangers =
+      [
+        &AstParser.parse/1,
+        &AstParserFullDetections.parse/1
+      ]
+      |> Enum.map(&apply(&1, [ast]))
```

Comment:
IMO this is easier to read. It's also somewhat easier to maintain if these functions ever change.
```suggestion
      [
        AstParser.parse(ast),
        AstParserFullDetections.parse(ast)
      ]
```